source/reference/code-blocks.txt

```diff
@@ -0,0 +1,467 @@
+
+   .. code-block:: <language>
+      :copyable: <boolean>
+      :emphasize-lines: <line numbers>
+      :linenos:
+      :caption: <caption text>
+      :source: <url>
+
+      <code sample>
+
+Options
+~~~~~~~
+
+All of the following options are optional.
+
+.. list-table::
+
+   * - ``<language>``
+     - Language to use for syntax highlighting. 
+       
+       For a complete list of supported languages, see the
+       `leafygreen-ui GitHub repository. <https://github.com/mongodb/leafygreen-ui/tree/main/packages/code#properties>`__
+
+       **Default:** None
+ 
+   * - ``:copyable: <boolean>``
+     - Option that indicates whether to show the :guilabel:`copy icon`.
+
+       **Default:** ``true``
+
+   * - ``:emphasize-lines: <line number>``
+     - Lines to highlight. You can specify a range 
+       of line numbers, individual line numbers, or both.
+
+       **Default:** None
+
+   * - ``:linenos:``
+     - Option that indicates whether to show line numbers.
+
+       **Default:** None
+  
+   * - ``:caption: <caption text>``
+     - Brief description.
+
+       **Default:** None 
+
+   * - ``:source: <url>``
+     - Option that indicates whether to show the :guilabel:`source icon` and
+       :guilabel:`View full source` tooltip.
+       
+       This icon is hidden if the ``copyable`` option is set to ``false``.
+       
+       **Default:** None
+
+Example 1
+~~~~~~~~~
+
+.. code-block::
+   
+   .. code-block:: python
+      :linenos:
+      :copyable: true
+      :emphasize-lines: 1, 4-5
+      :caption: An example Python code block
+
+      print("Hello Docs team!") 
+      print("This is Python code...") 
+      print("...inside a code-block.")
+      print("The line numbers are over here <-")
+      print("And the copy icon is over there ->")
+
+The previous ``code-block`` directive creates the following output:
+
+.. code-block:: python
+   :linenos:
+   :copyable: true
+   :emphasize-lines: 1, 4-5
+   :caption: An example Python code block
+
+   print("Hello Docs team!") 
+   print("This is Python code...") 
+   print("...inside a code-block.")
+   print("The line numbers are over here <-")
+   print("And the copy icon is over there ->")
+
+Example 2
+~~~~~~~~~
+
+.. code-block::
+   
+   .. code-block:: python
+      :source: https://github.com/mongodb/docs-ecosystem/blob/36f9cc7260a246d47cb05ac35276a2c92734b028/conf.py#L120
+
+      html_theme = sconf.theme.name
+      html_theme_path = [os.path.join(conf.paths.buildsystem, 'themes')]
+      html_title = conf.project.title
+      htmlhelp_basename = 'MongoDBdoc'
+
+The previous ``code-block`` directive creates the following output:
+
+.. code-block:: python
+   :source: https://github.com/mongodb/docs-ecosystem/blob/36f9cc7260a246d47cb05ac35276a2c92734b028/conf.py#L120
+
+      html_theme = sconf.theme.name
+      html_theme_path = [os.path.join(conf.paths.buildsystem, 'themes')]
+      html_title = conf.project.title
+      htmlhelp_basename = 'MongoDBdoc'
```

Comment:
I think for consistency, the attribute indentation should match the column that contains the `:source:` attribute even if the parser accepts this.